src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -69,6 +84,186 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         {
             _lastStartingDirectoryPath = StartingDirectory();
         }
+
+        // generate the font list, if we don't have one
+        if (!_FontList || !_MonospaceFontList)
+        {
+            UpdateFontList();
+        }
+    }
+
+    // Method Description:
+    // - Updates the lists of fonts and sorts them alphabetically
+    void ProfileViewModel::UpdateFontList() noexcept
+    try
+    {
+        // initialize font list
+        std::vector<Editor::Font> fontList;
+        std::vector<Editor::Font> monospaceFontList;
+
+        // get a DWriteFactory
+        com_ptr<IDWriteFactory> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(
+            DWRITE_FACTORY_TYPE_SHARED,
+            __uuidof(IDWriteFactory),
+            reinterpret_cast<::IUnknown**>(factory.put())));
+
+        // get the font collection; subscribe to updates
+        com_ptr<IDWriteFontCollection> fontCollection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(fontCollection.put(), TRUE));
+
+        for (UINT32 i = 0; i < fontCollection->GetFontFamilyCount(); ++i)
+        {
+            try
+            {
+                // get the font family
+                com_ptr<IDWriteFontFamily> fontFamily;
+                THROW_IF_FAILED(fontCollection->GetFontFamily(i, fontFamily.put()));
+
+                // get the font's localized names
+                com_ptr<IDWriteLocalizedStrings> localizedFamilyNames;
+                THROW_IF_FAILED(fontFamily->GetFamilyNames(localizedFamilyNames.put()));
+
+                // construct a font entry for tracking
+                const auto fontEntry{ _GetFont(localizedFamilyNames) };
+
+                // check if the font is monospaced
+                try
+                {
+                    com_ptr<IDWriteFont> font;
+                    THROW_IF_FAILED(fontFamily->GetFirstMatchingFont(DWRITE_FONT_WEIGHT::DWRITE_FONT_WEIGHT_NORMAL,
+                                                                     DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_NORMAL,
+                                                                     DWRITE_FONT_STYLE::DWRITE_FONT_STYLE_NORMAL,
+                                                                     font.put()));
+
+                    // add the font name to our list of monospace fonts
+                    const auto castedFont{ font.as<IDWriteFont1>() };
+                    if (castedFont && castedFont->IsMonospacedFont())
+                    {
+                        monospaceFontList.emplace_back(fontEntry);
+                    }
+                }
+                CATCH_LOG();
```

Comment:
This isn't gonna log 100 warnings as we iterate over all the fonts, is it?

Reply:
switched from `.as` to `.try_as`. So I think the only one we are really logging here is `GetFirstMatchingFont`. I think that's fair because we'd want to know which fonts we can't get the `IDWriteFont` version of?